lib/entities/entity_factory.cpp

```diff
@@ -300,12 +243,83 @@ std::pair<int, int> EntityFactory::createButton(const Button button, const doubl
     return std::make_pair(button_id, text_id);
 }
 
-int EntityFactory::createText(std::string text, int x, int y, int x_scale, int y_scale) {
+int EntityFactory::createText(std::string text, int x, int y, int x_scale, int y_scale, int font_size, Color color) {
     auto dst = std::unique_ptr<Rect>(new Rect{ 0, 0 , 0, 0});
-    auto r_text = renderableFactory.createText(text, 50, { 0, 255, 0, 255 }, (int)Layers::UI, std::move(dst));
+    auto r_text = renderableFactory.createText(text, font_size, color, (int)Layers::UI, std::move(dst));
     auto comps = std::make_unique<std::vector<std::unique_ptr<Component>>>();
     comps->push_back(std::make_unique<TransformComponent>(x, y, x_scale, y_scale, Direction::POSITIVE, Direction::POSITIVE));
     comps->push_back(std::make_unique<TextureComponent>(std::move(r_text)));
 
     return entityManager->createEntity(std::move(comps));
-}
\ No newline at end of file
+}
+
+int EntityFactory::createCharacterSelector(int player_id, int x, int y) {
+    auto dst = std::unique_ptr<Rect>(new Rect{ 0, 0 , 0, 0});
+    auto comps = std::make_unique<std::vector<std::unique_ptr<Component>>>();
+    comps->push_back(std::make_unique<CharacterSelectionComponent>(player_id));
+    comps->push_back(std::make_unique<TransformComponent>(x, y, 0, 0, Direction::POSITIVE, Direction::POSITIVE));
+
+    return entityManager->createEntity(std::move(comps));
+}
+
+void EntityFactory::changeCharacterSelectorTexture(int entity_id, Character character, bool create){
+    if(!create)
```

Comment:
I would add a comment which says that when create is true, the user has confirmed the characters and can be created in game if I understand correctly. I think it would make it more clear why the create boolean is send as argument and it's purpose.